system/cle/cle.c

```diff
@@ -82,6 +84,14 @@
 #  define COLOR_OUTPUT  VT100_GREEN
 #endif
 
+/* Poll timeout during getcursor,
+ * 50ms should be enough to fetch a single byte from powerful PC (or telent)
+ */
+
+#define CLE_POLL_TIMEOUT 50
+
+#define CLE_GETCURSOR_RETRY_COUNT 5
```

Comment:
Maybe 100ms + 10 tries so it will be up to 1s in total?
I'm not sure what the real use case numbers are.

Reply:
Maybe more, a response have about three bytes, will poll fd before get each byte.

So the 1s in your examples should be atleast 1s instead of up to 1s.